src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -12,15 +12,40 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class FileType extends AbstractType
 {
     /**
      * {@inheritdoc}
      */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['multiple']) {
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
+                $form = $event->getForm();
+                $data = $event->getData();
+
+                // submitted data for an input file (not required) without choosing any file
+                if (array(null) === $data) {
```

Comment:
So, what about using a `TextType` with a `multiple` option and expect it to properly map multiple `<input type="text">` and an array of strings?..

I'm just saying it's not how this form type is meant to be used. That's why the `CollectionType` one exists. It does not withdraw the legitimacy for the `FileType` to have a `multiple` option, as its html counterpart propose it.

Alright, let's move forwards, what would you suggest? Something like:

```diff
- if (array(null) === $data) {
+ if (array() === array_filter($data)) {
```

to detect empty data?